File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/adapter/holder/BaseViewHolder.java
package io.muun.apollo.presentation.ui.adapter.holder;

import android.view.View;
import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseViewHolder<T> extends RecyclerView.ViewHolder {

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public abstract void bind(T item);
}

File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/adapter/holder/BaseViewHolder.kt
package io.muun.apollo.presentation.ui.adapter.holder

import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    abstract fun bind(viewModel: T)

    open fun onItemClick() {
        // Override to apply visual/state changes upon click
    }
}
